onnx/shape_inference/implementation.cc

```diff
@@ -115,6 +115,42 @@ void checkShapesAndTypes(const TypeProto& inferredType, const TypeProto& existin
     fail_type_inference("type case unsupported. existing=", existingTypeCase, " inferred=", inferredTypeCase);
   }
 }
+// after symbolic shape_inference from inferredType, assign it to existingTensorType
+// TypeProto_Tensor or TypeProto_SparseTensor
+template <typename T>
+void generateSymbolicShape(T inferredType, T* existingTensorType, SymbolTable symbolTable) {
+  if (!inferredType.has_shape()) {
+    // No need to consider symbolic shape
+    *existingTensorType = inferredType;
+    return;
+  }
+  for (int i = 0; i < inferredType.shape().dim_size(); ++i) {
+    const auto& inferredDim = inferredType.shape().dim(i);
+    auto* typeAfterSymbolicShapeDim = existingTensorType->mutable_shape()->mutable_dim(i);
+    *typeAfterSymbolicShapeDim = inferredDim;
+    // set a symbol if it doesn't have dim_value and dim_param
+    if (!inferredType.shape().dim(i).has_dim_value() && !inferredType.shape().dim(i).has_dim_param()) {
+      typeAfterSymbolicShapeDim->set_dim_param(symbolTable.createNew());
+    }
+  }
+}
+
+const TypeProto* materializeSymbolicShapeShape(TypeProto* inferredType, SymbolTable symbolTable) {
```

Comment:
SymbolTable& symbolTable

Reply:
typo: materializeSymbolicShape**Shape**

Reply:
you want to update inferredType then why not just do inplace update? why are you returning const TypeProto? + 
symbolicInferredType == inferredType  (line 140) this is very confusing...